FizzBuzz/FizzBuzz/Program.cs

```diff
@@ -1,12 +1,226 @@
 ﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace FizzBuzz
 {
-    class Program
+    public class Flags
     {
+        public bool three = false;
+        public bool five = false;
+        public bool seven = false;
+        public bool eleven = false;
+        public bool thirteen = false;
+        public bool seventeen = false;
+    }
+
+    public class Program
+    {
+        private Flags setFlags(string[] args)
+        {
+            Flags flags = new Flags();
+            for (int i = 0; i < args.Length; i++)
+            {
+                if (args[i] == "3")
+                    flags.three = true;
+                else if (args[i] == "7")
+                    flags.seven = true;
+                else if (args[i] == "5")
+                    flags.five = true;
+                else if (args[i] == "11")
+                    flags.eleven = true;
+                else if (args[i] == "13")
+                    flags.thirteen = true;
+                else if (args[i] == "17")
+                    flags.seventeen = true;
+            }
+            return flags;
+        }
+
+        private string getIterations()
+        {
+            Console.WriteLine("Write maximum number : ");
+            string max;
+            max = Console.ReadLine();
+            return max;
+        }
+
+        public string FizzBuzzLogic(int i, Flags flags)

```

Comment:
In here, you could iterate through each rule and apply it to the string. You'll need to be careful in which order you apply the rules, though